geode-redis/src/main/java/org/apache/geode/redis/internal/executor/RedisResponse.java

```diff
@@ -52,59 +52,59 @@ public void afterWrite() {
   }
 
   public ByteBuf encode(ByteBufAllocator allocator) {
-    return coderCallback.apply(allocator);
+    return coderCallback.apply(allocator.buffer());
```

Comment:
At some point I want to push this allocation further up the stack. I believe this ends up being safe because we always pass the returned before to a netty method which will end up freeing it. But I would prefer if it was always clear when we allocate a buffer that it will be freed.